### PR TITLE
refactor: move cookbook to user-guide and reference

### PR DESCRIPTION
**Pull request by nicecui**
Move the cookbook to `/reference/sql/functions` and `user-guide/query-data/sql.md` for the following reasons:

- I think that the cookbook existed due to the lack of examples in querying data and the reduction of documentation on SQL functions.
- Documents would prefer to link to `reference` or `query-data` instead of linking to the cookbook. For example, linking to `reference/sql/functions.md` from `user-guide`, dashboard console, etc.
- Writing more and more examples in the cookbook over time is not a good idea because it requires organizing the content. I think the structure in the end would be the same as write-data or query-data.

Therefore, I moved the content to the expected place and removed the cookbook.

### CODE REVIEW

File: docs/reference/sql/functions.md [hunk 1]
```diff
@@ -11,5 +15,140 @@ functions in DataFusion. These functions include:
...
+arrow_cast(expression, datatype)
+```
+
+Where the `datatype` can be any valid Arrow data type in this [list](https://arrow.apache.org/datafusion/user-guide/sql/data_types.html). The four timestamp types are:
+- Timestamp(Second, None)
+- Timestamp(Millisecond, None)
+- Timestamp(Microsecond, None)
+- Timestamp(Nanosecond, None)
+
+(Notice that the `None` means the timestamp is timezone naive)
```
Comment (sunng87) [inline docs/reference/sql/functions.md:32]: Our http API output them as 

- Timestamp
- TimestampMillisecond
- TimestampMicrosecond
- TimestampNanosecond

I think we can use these names here avoid explaining `None`

File: docs/reference/sql/functions.md [hunk 2]
```diff
@@ -11,5 +15,140 @@ functions in DataFusion. These functions include:
...
+Please refer to [API documentation](https://greptimedb.rs/script/python/rspython/builtins/greptime_builtin/index.html#functions)
+
+## Time and Date
+
+### `date_trunc`
+
+`date_trunc` function follows the same API with PostgreSQL's [`date_trunc`](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC). It's illustrated as:
+
+```sql
+date_trunc(field, source [, time_zone ])
```
Comment (sunng87) [inline docs/reference/sql/functions.md:45]: rename `field` to a more meaningful name?

File: docs/reference/sql/functions.md (hunk 1 above)
Comment (nicecui) [reply docs/reference/sql/functions.md:32]: No, that's the function input, not an API output.

`arrow_cast` function doesn't support expression like `select * from monitor where ts > arrow_cast(1690252336408, 'TimestampMillisecond');`

The error would be `Error during planning: Unsupported type 'TimestampMillisecond'. Must be a supported arrow type name such as 'Int32' or 'Timestamp(Nanosecond, None)'. Error unrecognized word: TimestampMillisecond`

File: docs/reference/sql/functions.md (hunk 2 above)
Comment (nicecui) [reply docs/reference/sql/functions.md:45]: In the PostgreSQL's doc:

> `field` selects to which precision to truncate the input value.

maybe we could change `field` to `precision`

**Comment by nicecui** [comment]
@sunng87 PTAL

**Review by sunng87 — APPROVED** [review]

File: docs/reference/sql/functions.md [hunk 3]
```diff
@@ -11,5 +15,140 @@ functions in DataFusion. These functions include:
...
+- month
+- quarter
+- year
+- decade
+- century
+- millennium
+
+### INTERVAL
+
+The interval data type allows you to store and manipulate a period of time in years, months, days, hours etc. It's illustrated as:
```
Comment (xtang) [inline docs/reference/sql/functions.md:65]:
```suggestion
The Interval data type allows you to store and manipulate a period of time in years, months, days, hours etc. It's illustrated as:
```

**Review by xtang — COMMENTED** [review]
Almost LGTM, and PTAL at the comments

**Review by xtang — APPROVED** [review]

**Pull request merged**